backend/src/rss.rs

```diff
@@ -0,0 +1,250 @@
+
+#[derive(Debug)]
+struct Event {
+    id: Key,
+    title: String,
+    description: Option<String>,
+    created: DateTime<Utc>,
+    creators: Vec<String>,
+    thumbnail_url: Option<String>, 
+    tracks: Vec<EventTrack>,
+}
+
+impl_from_db!(
+    Event,
+    select: {
+        events.{ id, title, description, creators, thumbnail, created, tracks },
+    },
+    |row| {
+        Self {
+            id: row.id(),
+            title: row.title(),
+            description: row.description(),
+            created: row.created(),
+            creators: row.creators(),
+            thumbnail_url: row.thumbnail(),
+            tracks: row.tracks(),
+        }
+    }
+);
+
+/// Generates the xml for an RSS feed of a series in Tobira.
+pub(crate) async fn generate_feed(context: &Arc<Context>, id: &str) -> Result<String, Response> {
+    let db_pool = &context.db_pool;
+    let tobira_url = context.config.general.tobira_url();
+    let series_link = format!("{}/!s/{}", tobira_url, id);
+    let rss_link = format!("{}/~rss/series/{}", tobira_url, id);
+
+    let Some(series_id) = Key::from_base64(id) else {
+        return Err(bad_request("invalid series ID"));
+    };
+    
+    let db = db::get_conn_or_service_unavailable(db_pool).await?;
+
+    let query = "select opencast_id, title, description from series where id = $1";
+    let series_data = match db.query_one(query, &[&series_id]).await {
+        Ok(data) => data,
+        Err(_) => return Err(bad_request("DB error querying series data")),
+    };
+
+    let series_oc_id = series_data.get::<_, String>("opencast_id");
+    let series_title = series_data.get::<_, String>("title").clone();
+    let series_description = series_data.get::<_, Option<String>>("description");
+    let cover_image = context.config.theme.logo.large.path.clone();
+
+    // Build rss xml
+    let mut xml = XMLBuilder::new()
+        .version(XMLVersion::XML1_0)
+        .encoding("UTF-8".into())
+        .build();
+
+    let mut rss = XMLElement::new("rss");
+
+    // Add rss attributes
+    let attributes = [
+        ("version", "2.0"),
+        ("xmlns:dc", "http://purl.org/dc/elements/1.1/"),
+        ("xmlns:content", "http://purl.org/rss/1.0/modules/content/"),
+        ("xmlns:atom", "http://www.w3.org/2005/Atom"),
+        ("xmlns:media", "http://search.yahoo.com/mrss/"),
+        ("xmlns:itunes", "http://www.itunes.com/dtds/podcast-1.0.dtd"),
+    ];
+    for (name, value) in attributes {
+        rss.add_attribute(name, value);
+    }
+
+    let mut channel = XMLElement::new("channel");
+
+    // Add channel tags
+    let channel_data = [
+        ("title", vec![("", series_title.clone())]),
+        ("link", vec![("", series_link)]),
+        ("description", vec![("", series_description.unwrap_or_default())]),
+        ("language", vec![("", "und".to_string())]),
+        ("itunes:explicit",vec![("", "true".to_string())]),
+        ("itunes:image", vec![("href", (&cover_image.to_string_lossy()).to_string())]),
```

Comment:
Noticed this by chance just now: this prints the path on the machine, not the public URL.